design_docs/oscode/oscode_sundials.md

```diff
@@ -0,0 +1,375 @@
+
+$$
+\delta x_{ARDC} = A_+ \delta f_+ + A_- \delta f_-
+$$
+$$
+\delta f_\pm = f_\pm \sum_{i=0}^n \delta[S_i]_t^{t + h}
+$$
+
+$$
+\delta \dot{x}\_{ARDC} = B\_+ \delta \dot{f}\_+ + B\_- \delta \dot{f}\_-
+$$
+
+$$
+\delta \dot{f}\_\pm = \delta f\_\pm \frac{\dot{f}\_\pm}{f\_\pm}
+$$
+
+OSCODE's time stepping modules may be run in a variety of "modes":
+
+- NORMAL: The solver will take internal steps until it has just overtaken a user-specified output time, $t_{out}$, in the direction of integration, i.e. $t_{n-1} < t_{out} \le t_n$ for forward integration, or $t_{n} < t_{out} \le t_{n-1}$ for backward integration. It will then compute an approximation to the solution $y(t_{out})$ by interpolation (using one of the dense output routines described in the section TBD).
+
+- ONE-STEP: The solver will only take a single internal step $y_{n-1} \rightarrow y_n$ and then return control back to the calling program. If this step will overtake $t_{out}$ then the solver will again return an interpolated result; otherwise it will return a copy of the internal solution $y_n$
+
+- NORMAL-TSTOP: The solver will take internal steps until the next step will overtake $t_{out}$. It will then limit this next step so that $t_n = t_{n-1} + h_n = t_{out}$, and once the step completes it will return a copy of the internal solution
+
+- ONE-STEP-TSTOP: The solver will check whether the next step will overtake $t_{out}$ if not then this mode is identical to "one-step" above; otherwise it will limit this next step so that $t_n = t_{n-1} + h_n = t_{out}$. In either case, once the step completes it will return a copy of the internal solution $y_n$.
+.
+
+We note that interpolated solutions may be slightly less accurate than the internal solutions produced by the solver. Hence, to ensure that the returned value has full method accuracy one of the "tstp" modes may be used.
+
+### Interpolation
+
+As mentioned above, the time-stepping modules in OSCODE support interpolation of solution $y(t_{out})$ and derivatives $y^{(d)}(t_{out})$, where $t_{out}$ occurs within a completed time step from $t_{n-1} \rightarrow t_n$. Additionally, this module supports extrapolation of solutions and derivatives for outside this interval (e.g. to construct predictors for iterative nonlinear and linear solvers). To this end, OSCODE currently supports construction of polynomial interpolants $p_q(t)$ of polynomial degree up to $q=5$, although users may select interpolants of lower degree.
+
+OSCODE provides three complementary interpolation approaches, both of which are accessible from any of the time-stepping modules: "Hermite" and "Lagrange". The Hermite approach has been included with ARKODE since its inception, and is more suitable for non-stiff problems; Lagrange is designed to provide increased accuracy when integrating stiff problems. These two methods are described in detail within the ARKODE documentation (link).
```

Comment:
Was this a typo or is there a third option?

```suggestion
OSCODE provides two complementary interpolation approaches, both of which are accessible from any of the time-stepping modules: "Hermite" and "Lagrange". The Hermite approach has been included with ARKODE since its inception, and is more suitable for non-stiff problems; Lagrange is designed to provide increased accuracy when integrating stiff problems. These two methods are described in detail within the ARKODE documentation (link).
```